applications/MeshMovingApplication/custom_utilities/move_mesh_utilities.cpp

```diff
@@ -44,21 +45,45 @@ void CheckJacobianDimension(GeometryType::JacobiansType &rInvJ0,
 
 //******************************************************************************
 //******************************************************************************
+
 void MoveMesh(const ModelPart::NodesContainerType& rNodes) {
     KRATOS_TRY;
 
     const int num_nodes = rNodes.size();
     const auto nodes_begin = rNodes.begin();
 
-    #pragma omp parallel for
-    for (int i=0; i<num_nodes; i++) {
-        const auto it_node  = nodes_begin + i;
-        noalias(it_node->Coordinates()) = it_node->GetInitialPosition()
-            + it_node->FastGetSolutionStepValue(MESH_DISPLACEMENT);
-    }
+    IndexPartition<size_t>( num_nodes ).for_each(
+        [&]( size_t index )
+        {
+            const auto it_node  = nodes_begin + index;
+            noalias(it_node->Coordinates()) = it_node->GetInitialPosition()
+                + it_node->FastGetSolutionStepValue(MESH_DISPLACEMENT);
+        }
+    );
+
     KRATOS_CATCH("");
 }
 
+
+// block_for_each discards const qualifiers so this won't work
+/*
+void MoveMesh(const ModelPart::NodesContainerType& rNodes) {
+    KRATOS_TRY;
+
+    using Node = typename ModelPart::NodeType;
+
+    block_for_each( rNodes,
+        []( const Node& node )
+        {
+            noalias(node.Coordinates()) = node.GetInitialPosition()
+                + node.FastGetSolutionStepValue(MESH_DISPLACEMENT);
+        }
+    );
+
+    KRATOS_CATCH("");
+}
```

Comment:
BlockPartition expects a non-const reference for construction, so it can't be used here in it's current state. @philbucher Can I try modifying BlockPartition so it can be constructed from const and non-const objects or are const objects deliberately not allowed?

Alternatively, I can just keep using IndexPartition for const cases.

Reply:
Ah good to know

Yes you can try! Lets see what it takes, thx

Reply:
Just updating this PR with what we talked about for completeness' sake:

```MoveMesh``` actually does modify it's argument even though it's a const reference, so there's no need to come up with a const version of ```block_for_each``` yet.